lib/std/process.zig

```diff
@@ -299,152 +301,223 @@ pub const ArgIteratorWasi = struct {
     }
 };
 
-pub const ArgIteratorWindows = struct {
-    index: usize,
-    cmd_line: [*]const u16,
+pub fn ArgIteratorGeneral(comptime options: anytype) type {
+    return struct {
+        allocator: *Allocator,
```

Comment:
The original implementation didn't need an allocation.  My guess is you need one to implement string escapes?  My suggestion would be to leave the allocator out of the iterator, and allow the caller to perform the escape/allocation if desired.

Reply:
Not exactly... The original implementation _did_ require an allocator. It was in the 'next()' call. I moved the allocator to the init() function to harmonize the interface cross-platform.